File: app/src/main/java/com/nan/xarch/util/Logger.java
package com.nan.xarch.util;

import android.util.Log;

import com.nan.xarch.BuildConfig;

import java.util.Locale;

public final class Logger {

    private static final String TAG = "Logger";

    private Logger() {
        throw new RuntimeException("No instances!");
    }

    private static boolean isLoggable() {
        return BuildConfig.DEBUG;
    }

    public static void i(String tag, String fmt, Object... args) {
        if (isLoggable()) {
            Log.i(tag, format(fmt, args));
        }
    }

    public static void d(String tag, String fmt, Object... args) {
        if (isLoggable()) {
            Log.d(tag, format(fmt, args));
        }
    }

    public static void w(String tag, String fmt, Object... args) {
        if (isLoggable()) {
            Log.w(tag, format(fmt, args));
        }
    }

    public static void e(String tag, String fmt, Object... args) {
        if (isLoggable()) {
            Log.e(tag, format(fmt, args));
        }
    }

    public static void e(String tag, Throwable t, String fmt, Object... args) {
        if (isLoggable()) {
            Log.e(tag, format(fmt, args), t);
        }
    }

    private static String format(String fmt, Object... args) {
        if (args.length == 0) {
            return fmt;
        } else {
            return String.format(Locale.getDefault(), fmt, args);
        }
    }
}

File: app/src/main/java/com/nan/xarch/util/Logger.kt
@file:JvmName("Logger")

package com.nan.xarch.util

import android.util.Log
import com.nan.xarch.BuildConfig
import java.util.*

private fun isLoggable(): Boolean {
    return BuildConfig.DEBUG
}

fun i(tag: String, fmt: String, vararg args: Any) {
    if (isLoggable()) {
        Log.i(tag, format(fmt, *args))
    }
}

fun d(tag: String, fmt: String, vararg args: Any) {
    if (isLoggable()) {
        Log.d(tag, format(fmt, *args))
    }
}

fun w(tag: String, fmt: String, vararg args: Any) {
    if (isLoggable()) {
        Log.w(tag, format(fmt, *args))
    }
}

fun e(tag: String, fmt: String, vararg args: Any) {
    if (isLoggable()) {
        Log.e(tag, format(fmt, *args))
    }
}

fun e(tag: String, t: Throwable, fmt: String, vararg args: Any) {
    if (isLoggable()) {
        Log.e(tag, format(fmt, *args), t)
    }
}

private fun format(fmt: String, vararg args: Any): String {
    return if (args.isEmpty()) {
        fmt
    } else {
        String.format(Locale.getDefault(), fmt, *args)
    }
}